tests/e2e/device_crd/device_crd_suit_test.go

```diff
@@ -0,0 +1,166 @@
+/*
+Copyright 2019 The KubeEdge Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package device_crd
+
+import (
+	"bytes"
+	"io/ioutil"
+	"net/http"
+	"os/exec"
+	"path"
+	"path/filepath"
+	"runtime"
+	"testing"
+	"time"
+
+	"github.com/kubeedge/kubeedge/tests/e2e/utils"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+//context to load config and access across the package
+var (
+	ctx          *utils.TestContext
+	cfg          utils.Config
+	nodeSelector string
+	NodeName     string
+)
+var (
+	runController      = "cd ${GOPATH}/src/github.com/kubeedge/kubeedge/cloud; sudo nohup ./edgecontroller > edgecontroller.log 2>&1 &"
```

Comment:
in the future , we should  get rid of these commands and instead have them in a script which can be invoked using os.exec.

Reply:
+1 Since the existing e2e framework makes use of these commands to bring up the KubeEdge setup, i have also followed the same for consistency, but we should be looking to move these commands to a script